### ISSUE TITLE
Android: *PickListeners not called on main thread

### ISSUE THREAD

**Issue by westnordost** [tangrams/tangram-es#1365]
The methods called in the LabelPickListener, MarkerPickListener and FeaturePickListener are not called on the main (UI) thread, but on some GLThread, which is not documented and in my opinion also not desirable behavor since usually a "pick" of a feature, label or marker should do something visual on the screen, just as the callback made on a tap of a button is made on the UI thread.

Notably, the callback is even made on the GLThread when  `mapController.pick*` was called from the UI thread, i.e. the standard case:
```
	@Override public boolean onSingleTapConfirmed(float x, float y)
	{
		// I am on the UI thread
		controller.pickLabel(x,y);
		return true;
	}

	@Override
	public void onLabelPick(LabelPickResult labelPickResult, float positionX, float positionY)
	{
		// I am not on UI thread. This will crash the application.
		Toast.makeText(getContext(), "Crash me", Toast.LENGTH_SHORT).show();
	}
```
Actually, the above example does not even produce any error log on the console like it would normally happen when trying to do something UI on a non-UI-thread. Instead, it simply crashes with no further comment. I guess this has something to do with JNI/native code that can't be helped.

My suggested solution is to post the callback on the main thread by default.

**Comment by karimnaaji** [comment]
Hey @westnordost , 

Thanks, this is a fair suggestion. You're right that there should be consistency with the callee thread and callback thread. For now, the workaround for you would be to call `runOnUIThread` from the listener callback, but this is not convenient for end-user. 

I'll look into this shortly and make sure the UI thread is the one being called from the listener callbacks.

**Comment by matteblair** [comment]
I've been aware of this situation for a while and have avoided changing the behavior because not all callbacks originate from the UI thread or require synchronization with the UI thread. In those cases, we're imposing a small, but unnecessary delay and synchronization cost. The most flexible option, as I see it, is to allow the user to perform any necessary synchronization.

There are some benefits to consistently running callbacks on the UI thread. It's more convenient in the case where the callback performs some View operation, as you mentioned. It also prevents a long callback operation from delaying the rendering loop, which is possible with the current implementation.

Just sharing my two cents here.

**Comment by matteblair** [comment]
This is resolved by #1495 and will be available in the next release.

**Comment by matteblair** [comment]
This is available now in 0.7.0
